File: arcade/python/arcade-intro/03_Smooth Sailing/012_sortByHeight.py
def sortByHeight(a):
    '''
    Some people are standing in a row in a park. 
    There are trees between them which cannot be moved. 
    Your task is to rearrange the people by their heights
    in a non-descending order without moving the trees.
    '''
    withoutTrees = [i for i in a if i != -1]
    withoutTrees.sort()
    currentIndex = 0 
    for i in range(len(a)):
        if a[i] != -1:
            a[i] = withoutTrees[currentIndex]
            currentIndex += 1
    return a

print(sortByHeight([-1, 150, 190, 170, -1, -1, 160, 180]))

File: arcade/go/arcade-intro/03_Smooth Sailing/012_sortByHeight.go
import "sort"

func sortByHeight(a []int) []int {
	var people []int
	var result []int

	for _, item := range a {
		if item != -1 {
			people = append(people, item)
		}
	}

	sort.Ints(people)
	currentIndex := 0
	for _, item := range a {
		if item == -1 {
			result = append(result, -1)
		} else {
			result = append(result, people[currentIndex])
			currentIndex++
		}
	}

	return result
}
